### PR TITLE
31436: Maintain full, current information for Employers in GlueDB

### PR DESCRIPTION
**Pull request by zacharylandes**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [31436](https://devops.dchbx.org/redmine/issues/31436) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @TreyE |

### CODE REVIEW

**Review by bill-transue — APPROVED** [review]
Updates the `EmployerEvents::EmployerImporter` to handle contacts, phones, and office locations and updates the `Employer` model to associate the new `EmployerContact` and `EmployerOfficeLocation` models where the information is stored. New specs are added for these new models and the original `EmployerImporter` spec  has been updated with the new contact elements to test the new features of the importer.

**Review by TreyE — CHANGES_REQUESTED** [review]
Whoops.  Employer contacts have emails.

File: app/models/employer_events/employer_importer.rb [hunk 1]
```diff
@@ -4,31 +4,111 @@ class EmployerImporter
...
       {
-        hbx_id: hbx_id,
-        fein: fein,
-        dba: dba,
-        name: company_name
+        hbx_id: @org.id,
+        fein: @org.fein,
+        dba: @org.dba,
+        name: @org.name
       }
```
Comment (TreyE) [inline app/models/employer_events/employer_importer.rb:24]: Sorry, this one can have nil values too - particularly dba.

File: app/models/employer_events/employer_importer.rb [hunk 2]
```diff
@@ -4,31 +4,111 @@ class EmployerImporter
...
+    end
+    
+    def new_phone(incoming_phone)
+      Phone.new(
+        phone_number:  incoming_phone.full_phone_number.last(7),
+        full_phone_number: incoming_phone.full_phone_number,
+        phone_type: incoming_phone.type,
+        primary: incoming_phone.is_preferred
+      )
+    end
```
Comment (TreyE) [inline app/models/employer_events/employer_importer.rb:114]: Careful here - last 4 of the phone number aren't always nil - sometimes there is an extension.

Also, glue doesn't usually even populate full phone number (!) it usually sticks that value in the "phone_number" field.  Makes perfect sense, right?

File: app/models/employer_events/employer_importer.rb [hunk 3]
```diff
@@ -4,31 +4,111 @@ class EmployerImporter
...
+    def new_address(incoming_address)
+      Address.new(
+        address_1: incoming_address.address_line_1,
+        address_2: incoming_address.address_line_2,
+        city: incoming_address.location_city_name,
+        state: incoming_address.location_state_code,
+        zip: incoming_address.postal_code,
+        address_type: incoming_address.type
+      )
+    end
```
Comment (TreyE) [inline app/models/employer_events/employer_importer.rb:130]: More nil checking needed here.

**Review by TreyE — CHANGES_REQUESTED** [review]
Please see notes about nils and how glue does weird things with the phone_number column.

File: app/models/employer_events/employer_importer.rb [hunk 4]
```diff
@@ -4,31 +4,118 @@ class EmployerImporter
...
+    
+    def new_phone(incoming_phone)
+      phone_attributes = {
+        phone_number: incoming_phone.full_phone_number, 
+        phone_type: incoming_phone.type,
+        primary: incoming_phone.is_preferred
+      }
+      phone_attributes.delete_if{ |k,v| v.blank?}
+      Phone.new(phone_attributes)
+    end
```
Comment (TreyE) [inline app/models/employer_events/employer_importer.rb:114]: What happens if there ISN'T a phone or email for an office location?

File: app/models/employer_events/employer_importer.rb [hunk 5]
```diff
@@ -4,31 +4,118 @@ class EmployerImporter
...
+        address_1: incoming_address.address_line_1,
+        address_2: incoming_address.address_line_2,
+        city: incoming_address.location_city_name,
+        state: incoming_address.location_state_code,
+        zip: incoming_address.postal_code,
+        address_type: incoming_address.type 
+      }
+      address_attributes.delete_if{ |k,v| v.blank?}
+      Address.new(address_attributes)
+    end
```
Comment (TreyE) [inline app/models/employer_events/employer_importer.rb:130]: Ditto here, what happens if there isn't an address?

**Review by TreyE — CHANGES_REQUESTED** [review]
One last little issue with the office locations.

**Review by TreyE — APPROVED** [review]
Issues corrected.

**Pull request merged**